src/components/mixins/DataFormat.js

```diff
@@ -0,0 +1,93 @@
+let Validator = require('validatorjs');
+
+// To include another language in the Validator with variable processmaker
+if (window.ProcessMaker && window.ProcessMaker.user && window.ProcessMaker.user.lang) {
+  Validator.useLang(window.ProcessMaker.user.lang);
+}
+
+export default {
+  props: {
+    dataFormat: {
+      type: String,
+      default() {
+        return 'string';
+      },
+    },
+  },
+  data() {
+    return {
+      validatorFormat: null,
+    };
+  },
+  watch: {
+    // Triggered whenever the v-model is updated
+    value() {
+      if (this.$parent.transientData) {
+        this.$parent.transientData[this.name] = this.formatValue(this.value);
+      }
+    },
+  },
+  methods: {
+    formatValueWith(value, format) {
+      this.$set(this, 'dataFormat', format);
+      return this.formatValue(value);
+    },
+    formatValue(value) {
+      if (!value && this.dataFormat !== 'boolean') {
+        return '';
+      }
+
+      //Rules
+      let rules = {
+        'int': 'integer',
+        'boolean': 'boolean',
+        'string': 'string',
+        'datetime': 'date',
+        'date': 'date',
+        'float': 'regex:/^[+-]?\\d+(\\.\\d+)?$/',
+        'currency': 'regex:/^\\d{1,3}(,\\d{3})*(\\.\\d\\d)?(\\D{0,3})$/',
+      };
+      //Validate rule format
+      this.validatorFormat = new Validator( {[this.name]: value}, {[this.name]: rules[this.dataFormat]}, null);
```

Comment:
Instead of using a comment, it will help with maintainability and be just as readable to refactor this block into a function called `validateRuleFormat`. Though the block below does more than just validation so a name like `formatValueIfValid` would make things more clear.

Reply:
the block was divided into two function one to validate and one to format